### ISSUE TITLE
Deployment resource parameters not correct

### ISSUE THREAD

**Issue by pierskarsenbarg** [pulumi/pulumi-azure-native#973]
The docs state that it `It can be a JObject or a well formed JSON string.` but getting the following error message during preview: `error: azure-native:resources:Deployment resource 'deployment' has a problem: 'properties.parameters' should be of type '' but got a string`

## Steps to reproduce

Following code will produce error:

```csharp
using Pulumi;
using Pulumi.AzureNative.Resources;
using Pulumi.AzureNative.Resources.Inputs;
using System.Text.Json;

class MyStack : Stack
{
    public MyStack()
    {
        // Create an Azure Resource Group
        var resourceGroup = new ResourceGroup("resourceGroup");

        var deployment = new Pulumi.AzureNative.Resources.Deployment("deployment", new DeploymentArgs
        {
            DeploymentName = "deploymenttest",
            Properties = new DeploymentPropertiesArgs
            {
                Mode = DeploymentMode.Incremental,
                Parameters = JsonSerializer.Serialize(new {foo = "bar", bar = "foo"})
            },
            ResourceGroupName = resourceGroup.Name,

        });
    }
}
```

Expected: Preview would successfully take place

Actual: Error message as stated above

**Comment by mikhailshilkov** [comment]
I agree it's not ideal but it has to be a `Dictionary<string, object>` right now

**Comment by thomas11** [comment]
Note that `azure-native:solutions:ApplicationDefinition` has the same issue, with one or more `json#/Any` parameters and the phrase _"It can be a JObject or well-formed JSON string"_ in the docs.

Maybe we should consider detecting this case in the code and adjusting the docs to the equivalent of `Dictionary<string, object>`.

**Comment by mikhailshilkov** [comment]
@danielrbradley Will that hold for all languages? I'd guess there would be some edge case that would compile and work with the old type but not the new one in some language(s), but I'm not sure.
